pkg/resources/azure/azure.go

```diff
@@ -161,16 +164,35 @@ func (g *resourceGetter) listVirtualNetworksAndSubnets(ctx context.Context) ([]*
 	return rs, nil
 }
 
-func (g *resourceGetter) toVirtualNetworkResource(vnet *network.VirtualNetwork) *resources.Resource {
+func (g *resourceGetter) toVirtualNetworkResource(vnet *network.VirtualNetwork) (*resources.Resource, error) {
+	var blocks []string
+	blocks = append(blocks, toKey(typeResourceGroup, g.resourceGroupName()))
+
+	nsgs := map[string]struct{}{}
```

Comment:
Why not use a `k8s.io/utils/set/set.Set[string]`?

Reply:
I think it's just an awkward way to not use memory. Should probably change to bool, to keep things simple?

Reply:
Interesting approach. Thanks.